src/menu-button.js

```diff
@@ -0,0 +1,75 @@
+import { html, LitElement } from 'lit-element';
+import menuButtonCSS from './styles/menu-button.js';
+
+class MenuButton extends LitElement {
+  static get styles() {
+    return menuButtonCSS;
+  }
+
+  static get properties() {
+    return {
+      icon: { type: String },
+      href: { type: String },
+      label: { type: String },
+      id: { type: String },
+      selected: { type: Boolean },
+      followable: { type: Boolean },
+    };
+  }
+
+  constructor() {
+    super();
+    this.icon = '';
+    this.href = '';
+    this.label = '';
+    this.id = '';
+    this.selected = false;
+    this.followable = false;
+  }
+
+  onClick(e) {
+    e.preventDefault();
+    this.dispatchMenuTypeSelectedEvent();
+  }
+
+  dispatchMenuTypeSelectedEvent() {
+    this.dispatchEvent(new CustomEvent('menuTypeSelected', {
+      bubbles: true,
+      composed: true,
+      detail: {
+        id: this.id,
+      },
+    }));
+  }
+
+  get buttonClass() {
+    return this.selected ? 'selected' : '';
+  }
+
+  get iconClass() {
+    return this.selected ? 'active' : '';
+  }
+
+  get menuItem() {
+    return html`
+      <span class="icon ${this.iconClass}">
+        ${this.icon}
+      </span>
+      <span class="label">${this.label}</span>
+    `;
+  }
+
+  render() {
+    return html`
+      <a
+        href="${this.href}"
+        class="menu-item ${this.buttonClass}"
+        @click=${this.followable ? undefined : this.onClick}
+      >
+        ${this.menuItem}
+      </a>
+    `;
+  }
+}
+
+customElements.define('menu-button', MenuButton);
```

Comment:
Yeah, I wasn't sure. The difficult part about needing internal components is that they need to be defined globally before you write them to the DOM, so there's no way to sandbox them within this component's shadow DOM. As a result, my thought was that if you were going to use this component, you have to have this sub component defined for it to work. Because of that, I opted to always have it defined.

Reply:
what did we do prior?

Reply:
I've used the same approach in ia-topnav. If you want ia-topnav, you want it to render, and that means registering all of the components.